File: SecondYearCS/OOP/Civitas/src/civitas/Casilla.java
package civitas;
import java.util.ArrayList;

public abstract class Casilla {

    //atributos de clase
    private String nombre;

    Casilla ( String nombre ) {
        this.nombre = nombre;
    }
    
    abstract void recibeJugador (int actual, ArrayList<Jugador> todos) throws Exception;

    boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        boolean res = false;
        if( actual >= 0 && todos.size() > actual ){
            res = true;
        }
        return res;
    }
    
    void informe(int actual, ArrayList<Jugador> todos){
        if( this.jugadorCorrecto(actual, todos) ){
            Diario.getInstance().ocurreEvento( "Jugador: " +
                     todos.get(actual).getNombre() + "\n " +  this.toString() );
        }
    }
    
    @Override
    public String toString(){
       String result = "Casilla: " + this.nombre;
       return result;
    }

    String get_nombre() {
        return nombre;
    }
    
    public TituloPropiedad getTituloPropiedad(){
        return null;
    }  
}

File: SecondYearCS/OOP/CivitasRB/lib/casilla.rb
require_relative 'sorpresa.rb'
require_relative 'tipo_sorpresa.rb'
require_relative 'titulo_propiedad.rb'
require_relative 'mazo_sorpresa.rb'
require_relative 'diario.rb'
require_relative 'dado.rb'
module CivitasRB
  class Casilla
    attr_accessor :nombre, :tituloPropiedad
    def initialize(nombre)
      @nombre = nombre
    end

   def jugadorCorrecto(act,todos)
      act >=0 && act < todos.length
    end
    
    def to_s
     res = "\nCasilla: " + @nombre
     return res
    end
    
    #Informar al diario de la ocurrencia 
    def informe(act,todos)
      if self.jugadorCorrecto(act,todos)
        Diario.instance.ocurre_evento("Jugador: " + 
                    todos.at(act).nombre + self.to_s)
      end
    end
    
    #LLamar al metodo paga impuesto de la clase jugador
    def recibeJugador(act,todos)
      raise NotImplementedError,
            "this #{self.class} doesn't respond to: "
    end

    def getTituloPropiedad
      return nil
    end
  end
end
